raft/log.go

```diff
@@ -220,6 +247,7 @@ func (l *Log) Open(path string) error {
 		return err
 	}
 	l.term = term
+	l.votedFor = 0
```

Comment:
The `votedFor` should always clear out whenever the term is changed. Re-opening in practice isn't common but re-opening in tests is.